fe/src/main/java/org/apache/doris/load/loadv2/dpp/SparkDpp.java

```diff
@@ -0,0 +1,848 @@
+                StructType dstTableSchema = DppUtils.createDstTableSchema(baseIndex.columns, false);
+                RollupTreeBuilder rollupTreeParser = new MinimalCoverRollupTreeBuilder();
+                RollupTreeNode rootNode = rollupTreeParser.build(etlTable);
+                System.out.println("Rollup Tree:" + rootNode);
+
+                for (EtlJobConfig.EtlFileGroup fileGroup : etlTable.fileGroups) {
+                    List<String> filePaths = fileGroup.filePaths;
+                    Dataset<Row> fileGroupDataframe = null;
+                    if (Strings.isNullOrEmpty(fileGroup.hiveTableName)) {
+                        fileGroupDataframe = loadDataFromFilePaths(spark, baseIndex, filePaths, fileGroup, dstTableSchema);
+                    } else {
+                        String taskId = etlJobConfig.outputPath.substring(etlJobConfig.outputPath.lastIndexOf("/") + 1);
+                        String dorisIntermediateHiveTable = String.format(EtlJobConfig.DORIS_INTERMEDIATE_HIVE_TABLE_NAME,
+                                tableId, taskId);
+                        fileGroupDataframe = loadDataFromHiveTable(spark, dorisIntermediateHiveTable, baseIndex, fileGroup, dstTableSchema);
+                    }
+                    if (fileGroupDataframe == null) {
+                        System.err.println("no data for file file group:" + fileGroup);
+                        continue;
+                    }
+                    if (!Strings.isNullOrEmpty(fileGroup.where)) {
+                        long originalSize = fileGroupDataframe.count();
+                        fileGroupDataframe = fileGroupDataframe.filter(fileGroup.where);
+                        long currentSize = fileGroupDataframe.count();
+                        unselectedRowAcc.add(currentSize - originalSize);
+                    }
+
+                    fileGroupDataframe = repartitionDataframeByBucketId(spark, fileGroupDataframe,
+                            partitionInfo, partitionKeyIndex,
+                            partitionKeySchema, partitionRangeKeys,
+                            keyColumnNames, valueColumnNames,
+                            dstTableSchema, baseIndex, fileGroup.partitions);
+                    processRollupTree(rootNode, fileGroupDataframe, tableId, etlTable, baseIndex);
+                }
+            }
+            spark.stop();
+        } catch (Exception exception) {
+            exception.printStackTrace();
+            System.err.println("spark dpp failed for exception:" + exception);
+            dppResult.isSuccess = false;
+            dppResult.failedReason = exception.getMessage();
+            dppResult.normalRows = scannedRowsAcc.value() - abnormalRowAcc.value();
+            dppResult.scannedRows = scannedRowsAcc.value();
+            dppResult.fileNumber = fileNumberAcc.value();
+            dppResult.fileSize = fileSizeAcc.value();
+            dppResult.abnormalRows = abnormalRowAcc.value();
+            throw exception;
```

Comment:
Are you sure you want to throw exception?

Reply:
yes, use exception to indicate the spark app failed.